server/data/comments.js

```diff
@@ -0,0 +1,36 @@
+const { ObjectId } = require('mongodb');
+const moment = require('moment');
+const { getProjectById } = require('./projects');
+const { projects } = require('../config/mongoCollections');
+const { isValidStr, internalServerErr } = require('../utils');
+
+const createComment = async (commentParam, user) => {
+	let { comment, projectId } = commentParam;
```

Comment:
let's validate the projectId, userId, and username as well. Basically all the data coming in
I know it is redundant, but Patrick hill...

Reply:
`

        let { comment, projectId } = commentParam;
        comment = isValidStr(comment, 'Comment');
	const projectCollection = await projects();
	const projectFind = await getProjectById(projectId);
`
Project Id has validation in getProjectById function. It will throw an Id not of type string error if it's double-validated. The same goes for userId validation. Either it should be validated in the route or data, otherwise, it will throw an error.